### DISCUSSION TITLE
List-KR DNS 핑 관련 문의

### DISCUSSION THREAD

**Discussion by relaxedbrownie** [List-KR/community#36] [Q&A] [answered]
### URL

‍

### FILTERS

‍https://adguardteam.github.io/HostlistsRegistry/assets/filter_25.txt

### Browser

(Other)

### Operating System

Apple iOS

### Ad-Blocker

AdGuard for iOS/iOS Pro

### Comment

사실 오차단은 아니고 글을 어디에 남겨야 할지 애매해서 일단 여기에 문의드립니다. DNS 필터로 'KOR: List-KR DNS'를 켠 상태로 폰을 쓰다 보면 지연 시간이 비정상적으로 길어지는 걸 확인할 수 있었어요.

<details><summary>Screenshot</summary>

![5F67BE8B-75F6-4A58-B6B3-CCED5ABC7A3B](https://user-images.githubusercontent.com/126370491/221391579-7ff4931d-4d69-4a74-b46d-4add0711c3cc.jpeg)

</details>

그래서 임시로 'KOR: List-KR DNS'를 끄고 기본 필터인 'AdGuard DNS filter'를 사용하니까 정상적으로 돌아오더군요.

<details><summary>Screenshot</summary>

![3AA69DEA-096A-41B5-8A81-AC60C84F988D](https://user-images.githubusercontent.com/126370491/221391781-748f0a1a-f5e2-4522-b5a0-495f992b8a7c.png)

</details>

혹시 우연일지도 몰라서 일주일 정도 기간을 잡고 세 번 왔다 갔다 하면서(변경할 때마다 통계 초기화함) 주기적으로 확인했는데 결과는 비슷했어요. 'KOR: List-KR DNS'를 사용하면 100~300ms가 나오고 'AdGuard DNS filter'를 사용하면 세 자리 수는 절대 나오지 않았어요.

애드가드 DNS 보호 설정에서 DNS 구현은 'AdGuard', DNS 서버는 '시스템 기본 설정 (일반)'으로 해둔 상태예요.

제가 여기에 대한 지식이 전무해서 DNS 서버가 같은데 필터만으로도 지연 시간이 달라진다는 점이 의아하긴 한데 일단 테스트 결과는 그렇습니다. 참고로 제 기기만 그런 게 아니라 해당 DNS 필터를 사용하는 제 친구의 아이폰에서도 평균 시간이 200ms쯤 찍혀있더라고요. 기본 필터는 규칙 수가 너무 과해서 'KOR: List-KR DNS'를 사용하고 싶은데 해당 문제를 고쳐주실 수 있다면 감사하겠습니다.

**Comment by seia-soto** [comment, marked as answer]
안녕하세요? iOS 16.3.1 AdGuard Pro 4.4.5 (900)에서 필터 목록에 List-KR DNS를 추가하여 실제로 지연 시간이 증가하는 것을 확인했습니다. 그래서 추가 확인을 위해 macOS Safari 개발자 도구에서 지연 시간을 확인했으나 표시되는 시간과는 일치하지 않았습니다. 애플리케이션 내부 구조를 알 수 없어 정확한 요인은 알 수 없지만 1차적으로는 앱 전환 중에 추가 지연 시간이 표시될 수 있다고 생각합니다.

또는 필터를 추가했을 때 지연 시간이 눈에 띄게 늘어나는 이유는 필터에 추가된 와일드카드 형식의 도메인 형식이 원인일 수 있습니다. 실제로 와일드카드 형식의 필터는 기계적으로 처리하기에 더 많은 컴퓨팅이 필요하기에 따로 필터에서 와일드카드 형식의 레코드를 모두 제거하여 사용하시면 도움이 되실 수 있습니다.

필터 수정과 관련해서는 와일드카드 도메인에 대한 정보가 저에게는 없어 저로서는 현재 큰 방도가 없습니다.

테스트하시면서 사용하시는 애플리케이션이나 웹 사이트 종류를 알려주시면 보다 같은 환경에서 테스트가 가능합니다.

**Reply by piquark6046** [reply]
잠시만요. AdGuard Filter maintainers의 리더에게 여쭤볼께요.

**Reply by piquark6046** [reply]
실제 웹 브라우징에서는 핑이 낮은가요? 아니면, 표시되는 것처럼 느려지나요?

**Reply by relaxedbrownie** [reply]
제가 이 속도에 익숙해져서 그런 건지는 모르겠지만 느려졌다는 체감은 안 됐어요. 애드가드 vpn을 끄고 비교해봐도 그게 그거 같고요. 이게 항상 그런 게 아니라 느려지는 순간이 따로 있는 걸 수도 있어서 좀 애매하네요.
![8B2A4B50-1204-40CF-8A4A-1131C680FE75](https://user-images.githubusercontent.com/126370491/221489707-9f114685-7c3c-40be-9933-dd9979dd426c.jpeg)
이게 도움이 될지 모르겠지만 한 번 올려볼게요.

**Reply by piquark6046** [reply]
AdGuard Software Ltd의 adguard-ios Slack 채널에 메시지를 남겼습니다.

**Reply by piquark6046** [reply]
이에 대한 답변은 DNS 필터에 와일드카드가 있는 경우 딜레이가 발생할 수 있으나, 의미가 있을 정도는 아니라고 합니다.
또한, AdGuard for iOS에 표시되는 지연 시간이 커지는 경우는 평균 시간 계산 버그나 활동 평균 시간 제출 [^1] 버그일 것 같다고 하네요.

[^1]: 원문: presentation

**Reply by relaxedbrownie** [reply]
그렇군요. 알아봐주셔서 감사합니다. 한 가지만 더 여쭤보고 싶은 게 있어요.
AdGuard for iOS에는 Safari 필터(Safari에서만 작동하는 필터)와 DNS 필터(모든 앱에서 광고를 차단하기 위한 필터)가 있는데 Safari 필터에 있는 List-KR의 설명을 보면 '필터링 향상을 위해 AdGuard 관련 규칙과 결합 및 보강되었습니다. 이 필터는 AdGuard 베이스 필터와 함께 사용하는 것이 좋습니다.'라고 적혀있어요. 그래서 저는 AdGuard 베이스 필터와 List-KR을 둘 다 활성화해서 사용 중이거든요. 혹시 DNS 필터에 있는 List-KR DNS도 AdGuard DNS filter(AdGuard for iOS에서 기본으로 제공하는 DNS 필터)와 함께 사용하는 것이 좋나요? 아니면 단독으로 사용해도 무방하게 만들어져있나요? DNS 필터의 List-KR DNS에는 Safari 필터의 List-KR과 달리 기본 필터와 함께 사용하라는 문구가 안 적혀있길래 여쭤봅니다.
![FB781DD0-DB54-47B6-A831-33063A985EB9](https://user-images.githubusercontent.com/126370491/221565870-7405658a-5edf-4127-a566-f6c2629b9d88.jpeg)

**Reply by piquark6046** [reply]
만약 Safari로 YouTube를 YouTube 프리미엄 계정과 광고 없이 시청하시길 원하시면, AdGuard 베이스 필터를 키실 수 밖에 없습니다.
DNS 필터는 한국어권 앱들의 광고만 차단하시길 원하시면, List-KR 필터 단독으로 사용하셔도 무방합니다.

**Reply by relaxedbrownie** [reply]
그럼 AdGuard DNS filter와 List-KR DNS를 둘 다 활성화해서 사용하는 경우에는 문제가 안 생기나요? 예를 들면 어떤 상황에서 두 필터가 서로 충돌 일으켜서 필터링 성능이 저하된다든지, 오류가 발생할 수도 있다든지요. 비슷한 유형(사실 비슷한지 아닌지도 저는 몰라요)의 필터는 하나만 사용하는 걸 추천한다고 어디서 본 거 같아 혹시나 해서 여쭤봐요. 이게 진짜 마지막 질문이에요.

**Reply by seia-soto** [reply]
안녕하세요? 필터의 갯수가 많아지면 성능에 잠재적으로 영향을 끼칠 수 있으나 충돌을 일으키거나 하지는 않습니다. 필터 문법에 우선순위를 지정할 수 있는 문법도 있으며 비슷한 유형의 필터를 사용하여서 여러 필터에 중복된 항목이 있더라도 중복되지 않고 하나만 적용될 것이기 때문에 괜찮습니다. 따라서 필터링의 품질과 성능 측면 모두에서 저하가 일어나지 않습니다.

**Reply by seia-soto** [reply]
Safari 필터의 경우에는 적용할 수 있는 필터의 기능과 갯수에 제한이 있습니다. 그래서 타 플랫폼과 다르게 필터의 적용에 있어서 비교적 자유롭지 못한 점이 있고 특히 모바일 환경에서는 필터의 갯수가 많아지면 웹 브라우징 속도 저하나 발열 등이 나타날 수 있기 때문에 PC와 비교했을 때 적은 필터를 유지하는 것에 이점이 있다고 볼 수 있습니다. 다만 어디까지나 광고 차단 그리고 배터리 효율 중 어느 것을 중시하는지는 사용자 분의 선택이므로 어떤 선택을 하더라도 장점만 있는 선택은 없다고 볼 수 있습니다. List-KR의 경우에는 주기적으로 관리되고 있기 때문에 오래되어 더 이상 사용되지 않는 필터가 있더라도 필터 업데이트를 통해 효율성 또한 개선할 수 있습니다. 감사합니다.

**Reply by relaxedbrownie** [reply]
친절한 답변 감사합니다. 궁금증은 다 해결됐어요.

**Reply by piquark6046** [reply]
AdGuard DNS 필터를 사용하실 때 제가 알고 있는 문제는 디시인사이드에서 글이 작성되지 않는 점이 대표적이라고 할 수 있겠네요.
각 DNS 필터가 서로 충돌을 일으키는 경우는 제가 알기로는 없을 거예요.

**Reply by relaxedbrownie** [reply]
맞아요. 가끔 디시인사이드에 글이 안 써지더라고요. 그게 AdGuard DNS filter 때문이었군요? 생각해보니 List-KR DNS를 단독 사용하고 난 뒤론 그 문제가 없었던 거 같아요.
제가 이 부분을 여쭤본 건 구글링하다가 어떤 커뮤니티에서 piquark님이 직접 댓글 다신 걸 본 적 있어요. List-KR과 YousList를 함께 사용하기보다는 List-KR만 사용하는 걸 추천한다고. 좀 오래돼서 이 내용이 맞는지는 모르겠네요.

**Reply by piquark6046** [reply]
AdGuard DNS 필터를 언제 안 사용하시게 된건지는 모르겠지만, 제 생각에서는 AdguardTeam/AdguardFilters#139787 이 병합된 이후로는 AdGuard 추적 보호 필터를 같이 사용하시면 AdGuard DNS 필터가 켜져 있어도 정상적으로 작동될 같다고 보고 있어요.

기글하드웨어라면 제가 맞고요.
YousList는 더 규칙량이 더 많은데 잘 관리되지 않고 있어서 비추천했어요. 기존에 uBlock Origin에서는 YousList를 한국어 필터 리스트 [^1]로 사용했는데, 몇 개월 전에 List-KR 필터 리스트로 변경되었거든요.

[^1]: AdGuard에서의 필터는 uBO에서의 필터 리스트와 같은 의미이고, AdGuard에서의 규칙은 uBO에서의 필터와 같은 의미입니다.

**Reply by relaxedbrownie** [reply]
필터를 더 추가하는데 오차단을 오히려 막아주는 건 신기하네요. 그런 문제는 저는 무조건 필터를 줄여야만 해결되는 줄 알았거든요. 오늘 어쨌거나 많은 도움이 되었습니다. 다음에 또 문의할 게 생기거나 보강해야 할 사이트를 발견하면 제보하러 올게요.

**Reply by piquark6046** [reply]
네. 감사합니다.
